3_run_rsync/tasks/rsync.yml

```diff
@@ -14,12 +14,60 @@
     mig_source_data_location_k8s_mount: "kubernetes.io~glusterfs"
 
 - set_fact:
-    mig_dest_service_url: "{{ svc.resources[0].status.loadBalancer.ingress[0].hostname }}"
+    mig_dest_service_url: "{{ rsync_route.resources[0].spec.host }}"
     mig_dest_data_location: "/mnt/{{ pvc_ns }}/{{ pvc_name }}"
     mig_source_data_location: "{{ mig_source_data_base_location }}/pods/{{ bound_pod_uid }}/volumes/{{ mig_source_data_location_k8s_mount }}/{{ volume_name }}/"
     mig_source_host: "{{ node_name }}"
     mig_dest_ssh_key_remote_location: "~/.ssh/dest_key"
 
+- delegate_to: "{{ mig_source_host|mandatory }}"
+  tags:
+    - sync_phase
+    - final_phase
+  block:
+    - name: "Install stunnel"
+      yum:
+        name: stunnel
+        state: latest
+      become: true
+
+    - name: "Write stunnel.conf"
+      copy:
+        content: |
+          foreground = no
+          pid =
+          sslVersion = TLSv1.2
+          client = yes
+          syslog = no
+          [ssh]
+          accept = 2222
+          CAFile = /etc/stunnel/tls.crt
+          cert = /etc/stunnel/tls.crt
+          connect = {{ mig_dest_service_url }}:443
+          verify = 2
+          key = /etc/stunnel/tls.key
+        dest: /etc/stunnel/stunnel.conf
+      become: true
+
+    - name: "Write tls.crt"
+      copy:
+        content: "{{ crt['content']|b64decode }}"
+        dest: /etc/stunnel/tls.crt
+      become: true
+
+    - name: "Write tls.key"
+      copy:
+        content: "{{ key['content']|b64decode }}"
+        dest: /etc/stunnel/tls.key
+        mode: 0600
+      become: true
+
+    - name: "Start stunnel"
+      systemd:
+        state: started
```

Comment:
Lets make this restarted instead of started. It will guarantee it restarts if they have to restart a transfer so it picks up the new config/certs.

Reply:
@jmontleon this is failing on test env worker nodes, might have to create systemd unit file and then run this, +1 on restarted